apps/distgo/cmd/dist/dist.go

```diff
@@ -30,11 +30,48 @@ import (
 	"github.com/palantir/godel/apps/distgo/cmd"
 	"github.com/palantir/godel/apps/distgo/cmd/build"
 	"github.com/palantir/godel/apps/distgo/params"
+	"github.com/palantir/godel/apps/distgo/pkg/imports"
 	"github.com/palantir/godel/apps/distgo/pkg/osarch"
 	"github.com/palantir/godel/apps/distgo/pkg/script"
 	"github.com/palantir/godel/apps/distgo/pkg/slsspec"
 )
 
+func RequiresDist(products []string, cfg params.Project, wd string) ([]string, error) {
+	buildSpecsWithDeps, err := build.SpecsWithDepsForArgs(cfg, products, wd)
+	if err != nil {
+		return nil, err
+	}
+	productsSlice := []string{}
+	productsMap := make(map[string]struct{})
+	for _, curSpecWithDeps := range buildSpecsWithDeps {
+		for _, curDistCfg := range curSpecWithDeps.Spec.Dist {
+			artifactPaths := FullArtifactsPaths(ToDister(curDistCfg.Info), curSpecWithDeps.Spec, curDistCfg)
+			for _, artifactPath := range artifactPaths {
+				if fi, err := os.Stat(artifactPath); err == nil {
+					if goFiles, err := imports.AllFiles(
+						path.Join(
+							curSpecWithDeps.Spec.ProjectDir,
+							curSpecWithDeps.Spec.Build.MainPkg,
+						),
+					); err == nil {
+						if newerThan, err := goFiles.NewerThan(fi); err == nil && !newerThan {
+							// if the dist artifact for the product already exists and none of the source files for the
+							// product are newer than the dist artifact, consider spec up-to-date
+							continue
```

Comment:
Macro question: is it not possible to delegate to `build.RequiresBuild` instead of duplicating the logic here?